File: site/src/main/java/fr/lernejo/travelsite/binder/EngineBinder.java
package fr.lernejo.travelsite.binder;

import org.springframework.stereotype.Service;

import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.util.Objects;

@Service
public class EngineBinder {

    public void fetchResultsFromBackend() throws IOException {
        // Load file in classpath
        new String(Objects.requireNonNull(getClass().getResourceAsStream("countries.txt")).readAllBytes(), StandardCharsets.UTF_8)
            .lines()//.forEach(
                // request backend
            /*)*/;
    }
}

File: site/src/main/kotlin/fr/lernejo/travelsite/binder/EngineBinder.kt
package fr.lernejo.travelsite.binder

import org.springframework.stereotype.Service
import java.nio.charset.StandardCharsets

@Service
class EngineBinder {

    fun fetchResultsFromBackend() {
        javaClass.classLoader.getResourceAsStream("countries.txt")?.readAllBytes()
            ?.let {
                // Load file in classpath
                String(it, StandardCharsets.UTF_8)
                    .lines().stream()
                    .forEach {
                        // request backend
                    }
            }
    }
}
